include/xtensor/xcontainer.hpp

```diff
@@ -81,7 +84,10 @@ namespace xt
         void reshape(const shape_type& shape, const strides_type& strides);
 
         void transpose();
+
         void transpose(const std::vector<size_type>& permute);
+        void transpose(const std::vector<size_type>& permute, no_check check_policy);
+        void transpose(const std::vector<size_type>& permute, do_check check_policy);
```

Comment:
I'd prefer to have a single public function binding on private implementation, something like:
```cpp
template <class Tag = transpose_policy::no_check>
void transpose(const std::vector<size_type>& permutation, Tag tag = Tag())
{
    transpose_impl(permutation, tag);
}
```
Thus we don't increase the number of functions in the interface if we need to add more tags. Besides we only have one function to document and we can enumerate all the available tags in a single place.

As in the implementation, `permute` should be renamed in `permutation` .